File: ones/ones.py
#!/usr/bin/env python

import sys

def ones(i):
   r = 0
   while i > 0:
      if i % 10 == 1:
         r += 1
      i /= 10
   return r

for line in sys.stdin:
   i = int(line)
   r = 0
   while i > 0:
      r += ones(i)
      i -= 1
   print r

File: ones/ones.pl
#!/usr/bin/perl



while(<>) {
   $i = $_;
   $r = 0;
   while( $i > 0 ) {
      for( $j = $i; $j > 0; $j /= 10 ) {
         ++$r if($j % 10 == 1 );
      }
      --$i;
   }
   print "$r\n"
}
